covalent/_workflow/electron.py

```diff
@@ -322,12 +322,21 @@ def __call__(self, *args, **kwargs) -> Union[Any, "Electron"]:
         # Setting metadata for default values according to lattice's metadata
         # If metadata is default, then set it to lattice's default
         for k in self.metadata:
-            if (
-                k not in consumable_constraints
-                and k in DEFAULT_METADATA_VALUES
-                and self.get_metadata(k) is DEFAULT_METADATA_VALUES[k]
-            ):
-                self.set_metadata(k, active_lattice.get_metadata(k))
+            if k != "executor":
+                if (
+                    k not in consumable_constraints
+                    and k in DEFAULT_METADATA_VALUES
+                    and self.get_metadata(k) is DEFAULT_METADATA_VALUES[k]
+                ):
+                    self.set_metadata(k, active_lattice.get_metadata(k))
+            # if electron executor metadata is not empty string, then it should assigned with whatever it has defined.
+            # else, we need to set the electron executor to be of lattices', which will be of default.
+            # this case is not possible without having an empty executor string at first, so i had to change this also at line 550
+            else:
+                if self.get_metadata(k) != "":
```

Comment:
Hi @FyzHsn  For the question of whether am i sure to remove `DEFAULT_METADATA_VALUES["executor"]` .
yes, will explain what I faced. 
First I coded line 336 as `if self.get_metadata(k) != DEFAULT_METADATA_VALUES["executor"]`
And that worked fine for all cases, except for one case. That goes as follows: 
Consider covalent started with dask, And lattice executor changed to local as `@ct.lattice(executor="local")`. Now all the electrons are supposed to get local. Then I changed one electron executor to default(dask) as `@ct.electron(executor="dask")`. Now as electrons precede lattice, it is supposed to take dask. But it doesn't, according to line 336 I coded at first, as the newly changed executor is actually the default one, and that makes it fail the if condition and gets local instead!! This can be avoided if I have the default electron executor as an empty string, **then we can accomplish the case I mentioned, where we change the electron executor to default one** as per the new code `if self.get_metadata(k) != ""` . so now, even if we change the electron executor to default( here - dask), it will be changed. Considering this case, I had to change the default executor to empty string. Please let me know, if the explanation is not clear.